### PR TITLE
Enable regex flags manipulation

### PR DESCRIPTION
**Pull request by omri374**
## Change Description

As a user I would like to customize the regex flags used by Presidio, to be able to pass flags not defined initially.

Regex flags can be set on the `PatternRecognizer`'s constructor, or in the `RecognizerRegistry` constructor to support all existing `PatternRecognizer`s:

```python
from presidio_analyzer import AnalyzerEngine, RecognizerRegistry

import regex as re

registry = RecognizerRegistry(global_regex_flags=re.DOTALL | re.MULTILINE | re.IGNORECASE)
engine = AnalyzerEngine(registry=registry)
engine.analyze(...)
```

## Issue reference

This PR fixes issue #1029

## Checklist

- [X] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [X] I have signed the CLA (if required)
- [X] My code includes unit tests
- [X] All unit tests and lint checks pass locally
- [X] My PR contains documentation updates / additions if required

### CODE REVIEW

**Review by SharonHart — APPROVED** [review]

**Review by navalev — APPROVED** [review]

**Review by SharonHart — APPROVED** [review]

**Review by SharonHart — APPROVED** [review]

**Review by navalev — APPROVED** [review]

File: presidio-analyzer/presidio_analyzer/pattern_recognizer.py [hunk 1]
```diff
@@ -37,9 +39,9 @@ def __init__(
         deny_list: List[str] = None,
         context: List[str] = None,
         deny_list_score: float = 1.0,
+        global_regex_flags: Optional[int] = re.DOTALL | re.MULTILINE | re.IGNORECASE,
```
Comment (ducquangkstn) [inline presidio-analyzer/presidio_analyzer/pattern_recognizer.py:42]: This is pretty bad. You just change the default value from `re.DOTALL | re.MULTILINE` to `re.DOTALL | re.MULTILINE | re.IGNORECASE`.

This affect us as a client of this library.

Comment (omri374) [reply presidio-analyzer/presidio_analyzer/pattern_recognizer.py:42]: @ducquangkstn thanks for the feedback. This change allows you to have more customizability with regards to regex flags. Is this blocking you in any way?

Comment (ducquangkstn) [reply presidio-analyzer/presidio_analyzer/pattern_recognizer.py:42]: >  Is this blocking you in any way?

Actually, no. It just took me a while to figure out why the behavior is changed when pumping presidio version.

We (my company) are lucky that we have some unit tests. Not sure about other ppl.

**Pull request merged**